### DISCUSSION TITLE
Example for backup/restore

### DISCUSSION THREAD

**Discussion by maitredede** [topolvm/topolvm#993] [Q&A] [answered]
Hello,

I am using topolvm, and I would like to backup/restore my lvm volumes.

I have seen [some limitations](https://github.com/topolvm/topolvm/blob/main/docs/limitations.md) of topolvm about snapshots, but I have not yet understood them properly yet.

May it be possible to create a backup using [Velero with CSI](https://velero.io/docs/main/csi/) ?

**Comment by ushitora-anqou** [comment, marked as answer]
Hi.

I'm not familiar with Velero and, as far as I know, no one has tried using it with TopoLVM yet.

According to the document you pointed, it seems Velero saves VolumeSnapshot and VolumeSnapshotContent resources in one cluster and applies them in another. However, TopoLVM might not work this way because it only supports restoring from a VolumeSnapshot resource on the same node where the snapshot was taken. You might find this TopoLVM document helpful: https://github.com/topolvm/topolvm/blob/6550055d01ccd98078bad0c755b9580e28374f7b/docs/snapshot-and-restore.md

I'm not entirely sure about all this, so I recommend checking if TopoLVM works well with Velero on your own before you use them in production. If you can, sharing your results here would be really helpful for other users.

**Comment by toshipp** [comment]
This discussion has been closed due to inactivity. Please feel free to reopen this discussion (or open a new one) if you still need to discuss this topic. Thank you for your contribution.
